transport-msg.tex

```diff
@@ -150,9 +142,9 @@ \subsection{Messages Definition}\label{sec:Virtio Transport Options / Virtio Ove
 from the device. The driver request features at an index and the device answers
 with the features bits at 256*index.
 
-TODO: How should a device signal that there are no features at the requested
- index ? is it ok to just specify that 0 will be returned in that case ? can
- we expect the driver to always know how much features a device has ?
+If there are no features available at and after the requested index, the
+ERROR message can be used by the device to signal to the driver that there is
```

Comment:
As a general note, I recommend avoiding the passive voice whenever possible to make it clearer what part of the system is responsible for taking a specific action. In this case, I can interpret the requirement two ways. If the device is permitted to either send an all-zero response or an error I would reword this to say:

If there are no features available at the requested index or after the requested index the device SHOULD send an ERROR message to the driver to signal that there is no need to scan for features at or above the requested index.

If it the device is required to send this message change SHOULD to a MUST.

Reply:
Agree i will avoid the passive voice.
On a second thought on this specific example, I am not sure it is really a good idea to give several solutions here so i will remove the error possibility and stick to an all zero response here.